addons/mail/controllers/thread.py

```diff
@@ -105,6 +105,8 @@ def mail_message_post(self, thread_model, thread_id, post_data, context=None, **
                 kwargs["partner_emails"], kwargs.get("partner_additional_values", {})
             )]
         post_data["partner_ids"] = list(set((post_data.get("partner_ids", [])) + new_partners))
+        if "everyone" in special_mentions:
+            post_data["partner_ids"] = [channel_member.partner_id.id for channel_member in thread.channel_member_ids if channel_member.partner_id]
```

Comment:
This is currently replacing all mentions in `partner_ids` by channel members, but it would be possible (in public channel?) to have non-members users mentioned in the same message I think, which would then be lost.